aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,373 @@
+
+	log.Printf("[DEBUG] Waiting for Global Accelerator endpoint group (%s) availability", d.Id())
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for Global Accelerator endpoint group (%s) availability: %s", d.Id(), err)
+	}
+
+	return resourceAwsGlobalAcceleratorEndpointGroupRead(d, meta)
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	endpointGroup, err := resourceAwsGlobalAcceleratorEndpointGroupRetrieve(conn, d.Id())
+
+	if err != nil {
+		return fmt.Errorf("Error reading Global Accelerator endpoint group: %s", err)
+	}
+
+	if endpointGroup == nil {
+		log.Printf("[WARN] Global Accelerator endpoint group (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	listenerArn, err := resourceAwsGlobalAcceleratorEndpointGroupParseListenerArn(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("listener_arn", listenerArn)
+	d.Set("endpoint_group_region", endpointGroup.EndpointGroupRegion)
+	d.Set("health_check_interval_seconds", endpointGroup.HealthCheckIntervalSeconds)
+	d.Set("health_check_path", endpointGroup.HealthCheckPath)
+	d.Set("health_check_port", endpointGroup.HealthCheckPort)
+	d.Set("health_check_protocol", endpointGroup.HealthCheckProtocol)
+	d.Set("threshold_count", endpointGroup.ThresholdCount)
+	d.Set("traffic_dial_percentage", endpointGroup.TrafficDialPercentage)
+	if err := d.Set("endpoint_configuration", resourceAwsGlobalAcceleratorEndpointGroupFlattenEndpointDescriptions(endpointGroup.EndpointDescriptions)); err != nil {
+		return fmt.Errorf("error setting endpoint_configuration: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupParseListenerArn(endpointGroupArn string) (string, error) {
+	parts := strings.Split(endpointGroupArn, "/")
+	if len(parts) < 6 {
+		return "", fmt.Errorf("Unable to parse listener ARN from %s", endpointGroupArn)
+	}
+	return strings.Join(parts[0:4], "/"), nil
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(configurations []interface{}) []*globalaccelerator.EndpointConfiguration {
+	out := make([]*globalaccelerator.EndpointConfiguration, len(configurations))
+
+	for i, raw := range configurations {
+		configuration := raw.(map[string]interface{})
+		m := globalaccelerator.EndpointConfiguration{}
+
+		m.EndpointId = aws.String(configuration["endpoint_id"].(string))
+		m.Weight = aws.Int64(int64(configuration["weight"].(int)))
+
+		out[i] = &m
+	}
+
+	log.Printf("[DEBUG] Expand endpoint_configuration: %s", out)
+	return out
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupFlattenEndpointDescriptions(configurations []*globalaccelerator.EndpointDescription) []interface{} {
+	out := make([]interface{}, len(configurations))
+
+	for i, configuration := range configurations {
+		m := make(map[string]interface{})
+
+		m["endpoint_id"] = *configuration.EndpointId
```

Comment:
To prevent potential panics, this should use the AWS Go SDK provided conversion function:

```suggestion
		m["endpoint_id"] = aws.StringValue(configuration.EndpointId)
```